src/pubsub.c

```diff
@@ -677,25 +722,27 @@ NULL
     }
 }
 
-void channelList(client *c, sds pat, dict *pubsub_channels) {
-    dictIterator *di = dictGetIterator(pubsub_channels);
-    dictEntry *de;
+void channelList(client *c, sds pat, dict **pubsub_channels, unsigned int slot_cnt) {
```

Comment:
`slot_cnt` must be `CLUSTER_SLOTS` if cluster is enabled and this is shard channels. In all other cases, `slot_cnt` should be 1. Is that right?

Then maybe it is better to just have a flag as argument and compute `slot_cnt` inside this function?

```suggestion
void channelList(client *c, sds pat, dict **pubsub_channels, int is_sharded) {
    unsigned int slot_cnt = (is_sharded && server.cluster_enabled) ? CLUSTER_SLOTS : 1;
```